pdp-info/info.go

```diff
@@ -0,0 +1,156 @@
+package info
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type streamObj struct {
+	stream PipService_GetRequestsStreamServer
+	err    chan error
+}
+
+type streamsPool struct {
+	streams map[string][]streamObj
+	sync.RWMutex
+}
+
+func newStreamsPool() *streamsPool {
+	return &streamsPool{
+		streams: make(map[string][]streamObj),
+	}
+}
+
+func (p *streamsPool) Register(stream PipService_GetRequestsStreamServer) chan error {
+	cerr := make(chan error, 1)
+
+	req := &Request{Query: "register"}
```

Comment:
it is difficult to understand the register flow here.  why don't ask the PIP to send the register request in the first command ? Because it is the PIP want to register itself to PDP.

Reply:
Fixed